src/components/form-elements/checkbox/Checkbox.jsx

```diff
@@ -0,0 +1,236 @@
+  /**
+   * Sets whether the checkbox input is marked as required. This doesn't affect checkbox style.
+   * @example <Checkbox required />
+   */
+  required?: boolean,
+  /**
+   * Value of the checkbox input.
+   * @example <Checkbox value="1" />
+   */
+  value?: string,
+  ...
+};
+
+const Checkbox = ({
+  checked,
+  children,
+  className,
+  color = 'dark',
+  defaultChecked = false,
+  description,
+  disabled = false,
+  errorMessage,
+  id,
+  indeterminate = false,
+  invalid = false,
+  customLabelId,
+  required = false,
+  name,
+  onChange,
+  value,
+  ...props
+}: CheckboxPropsType) => {
+  const {current: checkboxId} = React.useRef(
+    id === undefined || id === '' ? generateRandomString() : id
+  );
+
+  const [isChecked, setIsChecked] = React.useState(
+    checked !== undefined ? checked : defaultChecked
+  );
+  const inputRef = React.useRef(null);
+
+  React.useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.indeterminate = indeterminate;
+    }
+  }, [inputRef, indeterminate]);
+
+  React.useEffect(() => {
+    setIsChecked(checked);
+  }, [checked]);
+
+  const onInputChange = React.useCallback(
+    e => {
+      setIsChecked(e.target.checked);
+      if (onChange) {
+        onChange(e);
+      }
+    },
+    [onChange]
+  );
+
+  const checkboxClass = cx('sg-checkbox', className, {
+    'sg-checkbox--disabled': disabled,
+    [`sg-checkbox--${String(color)}`]: color,
+  });
+
+  if (__DEV__) {
+    invariant(
+      !(errorMessage && !invalid),
+      `Using 'errorMessage' property has no effect when 'invalid' property is not set.`
+    );
+    invariant(
+      !(errorMessage && !children),
+      `Using 'errorMessage' property should be used along with 'children' property (label).`
+    );
+  }
+
+  const errorTextId = `${checkboxId}-errorText`;
+  const descriptionId = `${checkboxId}-description`;
+
+  const describedbyIds = React.useMemo(() => {
+    const ids = [];
+
+    if (invalid && errorMessage) {
+      ids.push(errorTextId);
+    }
+    if (description) {
+      ids.push(descriptionId);
+    }
+    return ids.join(' ');
+  }, [errorTextId, descriptionId, invalid, errorMessage, description]);
+
+  return (
+    <div {...props} className={checkboxClass}>
+      <div className="sg-checkbox__wrapper">
+        <div className="sg-checkbox__element">
+          <input
+            ref={inputRef}
+            className="sg-checkbox__input"
+            id={checkboxId}
+            type="checkbox"
+            checked={isChecked}
+            disabled={disabled}
+            name={name}
+            onChange={onInputChange}
+            required={required}
+            value={value}
+            aria-checked={indeterminate ? 'mixed' : isChecked}
+            aria-invalid={invalid ? true : undefined}
+            aria-describedby={describedbyIds}
+            aria-labelledby={customLabelId}
+          />
+          <CheckboxIcon checked={isChecked} indeterminate={indeterminate} />
```

Comment:
I think that CheckboxIcon component should be simple UI element without repeating the same interface as checkbox.

```
checked={isChecked} indeterminate={indeterminate}
```

is interface that doesn't belong to icon and CheckboxIcon should provide simple `type` prop